### ISSUE TITLE
SRS from input LAS not copied to output

### ISSUE THREAD

**Issue by maphew** [hobuinc/untwine#176]
`unwtine -i path/to/dir_of_las -o converted.copc.laz` creates a file that does not have a coordinate system defined. 

This means the output file cannot be used in https://viewer.copc.io/. I haven't tried any other COPC client (I'm new to this format).

Attached are the `pdal info` for the output and one of the input las files (they all have the same srs, EPSG:4326).
* [input_las_info.json](https://github.com/user-attachments/files/17512757/input_las_info.json)
* [output_copc_laz_info.json](https://github.com/user-attachments/files/17512758/output_copc_laz_info.json)

**Comment by maphew** [comment]
attempting to use assign srs option yields an error:

>untwine -i path\to\terra_las -o output_copc.laz --a_srs espg:4326 --stats
> Untwine Error: Could not import coordinate system 'espg:4326': (unknown reason).

**Comment by maphew** [comment]
ahh, srs argument is case sensitive and needs to upper case, `--a_srs EPSG:4326`. 

The result still can't be read in viewer.copc.io. I don't see a public repository for that so don't know where to file the bug, if it is a bug. The error there is _"cartesian has a NaN component"_

**Comment by connormanning** [comment]
One note is that you should generally not use EPSG:4326 for COPC.  Since the index is a Cartesian octree, you should avoid using EPSG:4326 since all sorts of strange things can happen from assuming polar coordinates are Cartesian.  You might try UTM, or EPSG:3857 for example.

Also, I don't think your data is actually EPSG:4326... your `input_las_info.json` has a native bbox that is not 4326:
```
        "bbox":
        {
          "maxx": 458589.2905,
          "maxy": 7101546.721,
          "maxz": 880.2668976,
          "minx": 457889.4155,
          "miny": 7100501.704,
          "minz": 770.6110976
        },
```

The `bbox['EPSG:4326']` value in the info output has been reprojected to 4326 for convenience but does not mean that the data is natively in EPSG:4326.  So telling Untwine to consider the data as 4326 with `a_srs` definitely seems wrong.

You can use `pdal info <myfile.laz> --metadata` to see what PDAL extracts for the SRS, it appears in multiple formats in that output but one instance is the the `comp_spatialreference` key.  As far as I know, Untwine will simply forward the SRS from your input.

You have posted some output *derived* from your data but posting an actual sample file will be the best way to get help.  Otherwise we'll just be guessing, your problem could be invalid SRS, unrecognized (but valid) SRS, geographic SRS when it should be Cartesian, some issue like the COPC bounds being computed oddly from malformed data points, or a variety of other issues.

So please provide a sample file that reproduces the issue if you cannot get things working properly.

**Comment by maphew** [comment]
Thanks for the feedback. Is there a way to downsample the input into something reasonable for sharing and testing?

I was suprised to see EPSG:4326 reported for the LAS, I expected UTM-N08.

**Comment by maphew** [comment]
output of `pdal info myfile.las --metadata` :
[input_las_info_metadata.json](https://github.com/user-attachments/files/17513357/input_las_info_metadata.json).  It seems to be UTM Zone N8 WGS-84 throughout?

**Comment by maphew** [comment]
Specifically assigning the EPSG code that `pdal ... --metadata` reports for the input.las creates an output.laz that viewer.copc.io is happy with.

`untwine -i ....\from_dji_terra\lidars\terra_las -o output_copc.laz --a_srs EPSG:32608`

After this project is published I can share the source data for future troubleshooting if you're interested. That is a number of weeks out.

**Comment by abellgithub** [comment]
I don't think there's a bug here. Is that correct? Can this be closed?

**Comment by maphew** [comment]
Thanks to your help I have a workflow that lets me create a COPC file that can be read in at least one client. So I'm not stuck.

I'm confused why "pdal info" and "pdal info --metadata" report different SRS from each other. In my situation the metadata one is the correct one. Arguably it's a bug that the wrong one is used by untwine, but maybe for other folk it's reversed.

**Comment by connormanning** [comment]
I think there are some clarifications to be made here.

> I'm confused why "pdal info" and "pdal info --metadata" report different SRS from each other

This doesn't appear to be what's happening.  `pdal info` (without metadata) doesn't report an SRS at all.  It gives you the bounds in your native UTM coordinate system, and also reprojects those bounds to EPSG:4326 for convenience.  It is *not* reporting that your data is in EPSG:4326.  All the PDAL info you've posted looks correct to me, and there is no PDAL output you've posted that indicates your data is anything other than UTM.

> Arguably it's a bug that the wrong one is used by untwine

This is also not what's happening - as there is no "wrong one" to choose from.  Your Untwine COPC output is still UTM, and you can see from the `bbox` output that the data is UTM (just judging by the coordinates themselves, since the SRS is not reported in this output).  However, you may notice that you can *not* see an EPSG:4326 reprojected bbox, which implies that there is no SRS written in the COPC file, since PDAL could not reproject to 4326 for you.  So your COPC output is *not* being reported as EPSG:4326, it just doesn't have an SRS.

My *guess* is that Untwine is not forwarding this SRS properly because it is written as geotiff keys rather than WKT, according to your metadata output.  That would be a real bug in Untwine, and would explain why when you explicitly set the SRS for Untwine, it works.

# TLDR
- your data is UTM, and PDAL recognizes and reports this correctly
- Untwine does *not* appear to be forwarding this SRS to the output (this would be the bug, if true)
- there is no point in your workflow where I see the SRS is being incorrectly reported as EPSG:4326

**Comment by maphew** [comment]
>  pdal info (without metadata) doesn't report an SRS at all. It gives you the bounds in your native UTM coordinate system, and also reprojects those bounds to EPSG:4326 for convenience.

Ah, thank you for the clarification. I saw "EPSG" in the output and interpreted that as reporting the data SRS. I didn't know that bounding box could be reported in a different coordinate system without a change in data.

The rest of your assessment also seems correct to me.

**Comment by abellgithub** [comment]
When you provide `a_srs` as an untwine option, you're asking to for that SRS to be written to the output. There is no transformation associated with it. We may have no spatial reference to transform from. If you need to transform your data, you should do that ahead of time.

It seems strange that you got no SRS in your output. I would have to look at the original file(s) in order to evaluate further.

**Comment by maphew** [comment]
I can supply original files for troublseshooting after the parent project is published (unknown when). Or, if there's someway of downsampling the LAS while preserving the metadata etc. I can provide something sooner. (For example I've used `gdal_translate --outsize 10% 10%` to successfully diagnose problem geotiff on the gdal-dev mailing list.)
